generators/excel-schema/main.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+import os
+import csv
+import openpyxl
+import sys
+from datetime import datetime
+
+OPENPYXL_MAJOR_VERSION = int(openpyxl.__version__[0])
+DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S%Z'
+# Minimum possible header.
+HEADER = (
+    'domain',
+    'entity',
+    'attribute',
+    'value',
+)
+
+def generate(file_name, domain):
+   # Creating a CSV writer that feeds to stdout.
+    w = csv.writer(sys.stdout)
+
+    # Write the header to start.
+    w.writerow(HEADER)
+
+    wb = openpyxl.load_workbook(file_name, use_iterators=True)
+    sheets = wb.get_sheet_names()
+    name = os.path.splitext(os.path.basename(file_name))[0]
+
+    for i, sheet_name in enumerate(sheets):
+        sheet = wb.get_sheet_by_name(sheet_name)
+        column_names =  _column_names(wb, sheet_name)
+        rows = sheet.iter_rows()
+
+        # Skip header row because we already collected the column names.
+        header = next(rows)
+
+        for j, cell in enumerate(header):
+            label = list(column_names)[j]
+            column_alpha = cell.column
+            w.writerow([
+                domain,
+                label,
+                'label',
+                label
+            ])
+            w.writerow([
+                domain,
+                label,
+                'sheet',
+                sheet.title
+            ])
+            w.writerow([
+                domain,
+                label,
+                'column',
+                column_alpha
+            ])
+
+def _column_names(workbook, sheet_name):
+    sheet = workbook.get_sheet_by_name(sheet_name)
+    first_row = next(sheet.iter_rows())
+
+    # Renamed attribute in version 2+
+    if OPENPYXL_MAJOR_VERSION > 1:
+        return [c.value for c in first_row]
+
+    return [c.internal_value for c in first_row]
+
+    # Fallback to indexes
+    return range(len(first_row))
```

Comment:
Remove redundant `return` statement (please always review code that you copy and paste).